monai/bundle/config_item.py

```diff
@@ -164,17 +165,22 @@ class ConfigComponent(ConfigItem, Instantiable):
     Subclass of :py:class:`monai.bundle.ConfigItem`, this class uses a dictionary with string keys to
     represent a component of `class` or `function` and supports instantiation.
 
-    Currently, two special keys (strings surrounded by ``_``) are defined and interpreted beyond the regular literals:
+    Currently, three special keys (strings surrounded by ``_``) are defined and interpreted beyond the regular literals:
 
-        - class or function identifier of the python module, specified by one of the two keys.
-            - ``"_target_"``: indicates build-in python classes or functions such as "LoadImageDict",
-                or full module name, such as "monai.transforms.LoadImageDict".
+        - class or function identifier of the python module, specified by ``"_target_"``,
+          indicating a build-in python class or function such as ``"LoadImageDict"``,
+          or a full module name, such as ``"monai.transforms.LoadImageDict"``.
+        - ``"_requires_"``: specifies reference IDs (string starts with ``"@"``) or ``ConfigExpression``
```

Comment:
1. I think this doc-string is slightly misleading, if a component use `@` to refer to another component, it already depends on another component to evaluate or initialize first.
For example: `"dataloader": {"dataset": "@dataset"}`.
2. This `_requires_` is only useful for the case that component A and component B are **independent**, but A must run after B is evaluated or initialized.
3. And I think this `_requires_` key is not only useful in `ConfigComponent`, for example, maybe expression A must run after expression B?
4. And maybe it can be `_require_` instead of `_requires_`?

Thanks.

Reply:
thanks, these are good points. after the discussions we want to make sure the 'import' happens early to omit 'requires' for it, and we don't support '_requries_' for configexpression for now (because that could be addressed by other config patterns such as a list of expression notations)